### PR TITLE
Add worker skeleton for cross model relations

### PR DESCRIPTION
**Pull request by wallyworld**
Skeleton and associated boilerplate for new remote relations worker.

QA: bootstrap lxd

### CODE REVIEW

File: cmd/jujud/agent/model/manifolds_test.go [hunk 1]
```diff
@@ -4,22 +4,28 @@
...
 
 type ManifoldsSuite struct {
-	testing.IsolationSuite
+	testing.BaseSuite
 }
 
 var _ = gc.Suite(&ManifoldsSuite{})
 
+func (s *ManifoldsSuite) SetUpTest(c *gc.C) {
+	s.SetInitialFeatureFlags(feature.CrossModelRelations)
```
Comment (axw) [inline cmd/jujud/agent/model/manifolds_test.go]: can you please do this in a separate test? i.e. check the standard set of manifolds, and then have another test which checks what the effect of setting the feature flag is

File: worker/remoterelations/remoterelations.go [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+// Wait implements worker.Worker.
+func (w *Worker) Wait() error {
+	return w.catacomb.Wait()
+}
+
+func (w *Worker) loop() error {
+	for {
+		// TODO(wallyworld)
+		if w.killed() {
```
Comment (axw) [inline worker/remoterelations/remoterelations.go]: This is going to spin in a tight loop. I think you should just bring the select up into this loop, and ditch the default case.

File: worker/remoterelations/shim.go [hunk 3]
```diff
@@ -0,0 +1,24 @@
+// Copyright 2016 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package remoterelations
+
+import (
+	"github.com/juju/errors"
```
Comment (axw) [inline worker/remoterelations/shim.go]: \n

**Comment by wallyworld** [comment]
$$merge$$

**Pull request merged**